x/ibc/02-client/types/tendermint/consensus_state.go

```diff
@@ -0,0 +1,90 @@
+package tendermint
+
+import (
+	"bytes"
+	"errors"
+
+	lerr "github.com/tendermint/tendermint/lite/errors"
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	ics23 "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment"
+	"github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/merkle"
+)
+
+var _ exported.ConsensusState = ConsensusState{}
+
+// ConsensusState defines a Tendermint consensus state
+type ConsensusState struct {
+	ChainID          string                `json:"chain_id" yaml:"chain_id"`
+	Height           uint64                `json:"height" yaml:"height"` // NOTE: defined as 'sequence' in the spec
+	Root             ics23.Root            `json:"root" yaml:"root"`
+	NextValidatorSet *tmtypes.ValidatorSet `json:"next_validator_set" yaml:"next_validator_set"` // contains the PublicKey
+}
+
+// ClientType returns Tendermint
+func (ConsensusState) ClientType() exported.ClientType {
+	return exported.Tendermint
+}
+
+// GetHeight returns the ConsensusState height
```

Comment:
Since these properties are exported, why do we need getters for them?

Reply:
to fulfill the `ConsensusState` interface